src/tools/rbd_mirror/image_replayer/BootstrapRequest.cc

```diff
@@ -194,6 +194,10 @@ void BootstrapRequest<I>::handle_get_client(int r) {
     derr << ": failed to retrieve client: " << cpp_strerror(r) << dendl;
     finish(r);
     return;
+  } else if (m_client.state == cls::journal::CLIENT_STATE_DISCONNECTED) {
+    dout(0) << ": client flagged disconnected, re-registering" << dendl;
+    unregister_client();
```

Comment:
So, do we want that after disconnect the mirror does not reconnect automatically?

Right now, if "disconnected" is detected during replaying, the image repayer stops (without "manual" flag), so it is restarted in max 30 second interval by pool Replayer. And in bootstrap we reregister journal client, so the restart succeeds.

Do you mean the bootstap always returns error here and the "start" is terminated, similarly to "remote image is non-primary or local image is primary" case? So a user intervention is required to manually start image replay. And then how user is supposed to do this? Don't we want an automatic restart, may be after some configurable time interval?